app/Harvest/Repositories/AbstractRepository.php

```
@@ -45,17 +45,26 @@ public function getRows(SpiceHarvesterHarvest $harvest, \DateTime $from = null,
         $records = $endpoint->listRecords($harvest->metadata_prefix, $from, $to, $harvest->set_spec);
 
         try {
-            $total = $records->getTotalRecordCount();
-
             foreach ($records as $record) {
                 $row = $this->getDataRecursively($record, $this->fieldMap);
                 yield $row[0];
             }
         } catch (MalformedResponseException $e) {
-            $total = 0;
```

Is a silent failure really the right thing to do here? My preference would be not to catch the exception here, but I don't know this code well enough.

---

to be honest. I don't know. I don't even know, why there is this try/catch block... I think it's fair to raise exception, if it doesn't work - @rastislav-chynoransky can you explain?

---

i think if there is an empty result set you can't call getTotalRecordCount on it

---

so it's an exception that should be caught in the new getTotal method, since we don't consider it being an error state that we present to the user

---

but getting data from the oai provider is quite slow already now, i'd suggest not splitting it into two successive calls but to return an object containing `Generator` and total count in it

---

it's like @rastislav-chynoransky wrote. this try-catch block was there to handle empty result. it will raise exception both for `getDataRecursively()` and `getTotalRecordCount()` without this try-catch - thus I applied for both.
returning empty set is absolutely OK and common scenario... no reason to store error. 
now it ends up nicely:
```
Spracovaných záznamov: 0
0 nových záznamov
0 bolo upravených
0 bolo zmazaných
0 bolo preskočených
0 zlyhalo.
```

---

As an outsider, it seems strange to me to raise an exception for calling `getTotalRecordCount()` on a empty resultset. I'd expect `0` or `null` (provided empty resulset ~= `[]`). What's the reason for it?

---

That's not in our codebase.
https://github.com/caseyamcl/phpoaipmh/blob/933b4beb1c5d0de675d553ad6f9e545e633ba73b/src/RecordIterator.php#L229